libretro/libretro-core.c

```diff
@@ -2346,15 +2568,15 @@ void retro_run(void)
       retro_load_ok=true;
       app_init();
       update_variables();
-      pre_main(RPATH);
+      pre_main();
       runstate = RUNSTATE_RUNNING;
       return;
    } 
    else if (runstate == RUNSTATE_LOADED_CONTENT)
    {
       /* Load content was called while core was already running, just do a reset with autostart */
       datasette_control(DATASETTE_CONTROL_STOP);
-      autostart_autodetect(RPATH, NULL, 0, AUTOSTART_MODE_RUN);
+      autostart_autodetect(autostartString, NULL, 0, AUTOSTART_MODE_RUN);

```

Comment:
This functionality if broken in this PR. On static platforms like Switch/Vita, this is used to trigger a reset+autostart when Load Content is selected while the core is running.
 
With this PR, the reset happens but the content is not loaded. I suspect you reset autostartString somewhere.

Note that on shared platforms like Linux, Windows etc., this code is never executed, because the core is unloaded  completely from memory and then reloaded from scratch.